docs/development/module/maps.md

```diff
@@ -0,0 +1,358 @@
+  `$event->addBehavior()` method.
+
+In all cases the behavior's `id` (as defined it its config entity YML) is used.
+
+### Behavior settings
+
+Some behaviors may require additional settings based on their context. Best
+practice is to include these in the map settings so that they are available in
+the behavior JavaScript in the following way:
+
+`const settings = instance.farmMapSettings.behaviors.mybehavior;`
+
+This can be accomplished in different ways, depending on how the behavior is
+being attached to the map.
+
+[Map types](#map-types) can add behavior settings to their `options` property.
+For example:
+
+```yaml
+langcode: en
+status: true
+dependencies:
+  enforced:
+    module:
+      - my_module
+id: mymaptype
+label: My Map Type
+description: "My module's custom map type."
+behaviors:
+  - mybehavior
+options:
+  behaviors:
+    mybehavior:
+      mysetting: True
+```
+
+Maps added as [render](#render-element) or [form](#form-element) elements can
+add behavior settings in their `#map_settings` property. For example:
+
+```php
+$build['mymap'] = [
+  '#type' => 'farm_map',
+  '#map_settings' => [
+    'behaviors' => [
+      'mybehavior' => [
+        'mysetting' => TRUE,
+      ],
+    ],
+  ],
+  '#behaviors' => [
+    'mybehavior',
+  ],
+];
+```
+
+Behaviors that are added via [render event](#render-events) subscribers can add
+settings at the same time:
+
+```php
+$event->addBehavior('mybehavior', ['mysetting' => TRUE]);
+```
+
+All of the above approaches will make the settings available in the behavior
+JavaScript in the same place.
+
+## Render events
+
+farmOS will trigger an event when a map is rendered. Modules can set up an
+event subscriber to perform additional logic at that time, such as adding
+behaviors.
+
+For example, to add a behavior to all maps in farmOS, add the following two
+files (replace `my_module` with the module name, and `mybehavior` with the
+behavior name):
+
+`my_module/my_module.services.yml`
+
+```yaml
+services:
+  my_module_map_render_event_subscriber:
+    class: Drupal\my_module\EventSubscriber\MapRenderEventSubscriber
+    tags:
+      - { name: 'event_subscriber' }
+```
+
+`my_module/src/EventSubscriber/MapRenderEventSubscriber`
+
+```php
+<?php
+
+namespace Drupal\my_module\EventSubscriber;
+
+use Drupal\farm_map\Event\MapRenderEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * An event subscriber for the MapRenderEvent.
+ */
+class MapRenderEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      MapRenderEvent::EVENT_NAME => 'onMapRender',
+    ];
+  }
+
+  /**
+   * React to the MapRenderEvent.
+   *
+   * @param \Drupal\farm_map\Event\MapRenderEvent $event
+   *   The MapRenderEvent.
+   */
+  public function onMapRender(MapRenderEvent $event) {
+    $event->addBehavior('mybehavior');
+  }
+
+}
+```
```

Comment:
Since the `farm_map` render element has its own documentation page now, I wonder whether it would be good to have a section showing how to use the element as a "prototype" (i.e. with a custom instantiator) also. It would be pretty easy to copy my example from [this commit](https://github.com/farmOS/farmOS/commit/c92fe90985b65e09ec92fc802273c0b959974e0c), but I can also add it as a follow-up if you prefer...

Reply:
Ah good point! I actually forgot about the custom instantiator functionality. :-D

If you wouldn't mind following up in a separate PR that would be great to add.

As I was writing this page I was also curious about including some information about how to build a behavior with Webpack, so that it could use more OpenLayers features if it wanted to. I've never done that myself, though, so figured it could be a potential follow-up as well.